### PR TITLE
fix: update executor parameter

### PR DESCRIPTION
**Pull request by OrangeSodahub**

### CODE REVIEW

File: annlite/executor.py [hunk 1]
```diff
@@ -58,8 +60,11 @@ def __init__(
             raise ValueError('Please specify the dimension of the vectors to index!')
 
         self.metric = metric
-        self.limit = limit
+        self.match_args = match_args
         self.include_metadata = include_metadata
+        self.limit = limit
+        if 'limit' in self.match_args:
+            self.limit = self.match_args.get('limit')
```
Comment (numb3r3) [inline annlite/executor.py:67]: Let's just keep `limit` in the arguments to overwrite `match_args`. 
```suggestion
        if limit:
            self.match_args.update('limit', limit)
```

File: annlite/executor.py [hunk 2]
```diff
@@ -262,7 +270,7 @@ def search(
                     'Please wait for the pending documents to be indexed.'
                 )
 
-            flat_docs.match(self._index, filter=search_filter, limit=limit)
+            flat_docs.match(self._index, limit=self.limit, **match_args)
```
Comment (numb3r3) [inline annlite/executor.py:273]:
```suggestion
            flat_docs.match(self._index, **match_args)
```

**Review by numb3r3 — CHANGES_REQUESTED** [review]

File: annlite/executor.py (hunk 1 above)
Comment (OrangeSodahub) [reply annlite/executor.py]: And the default value of limit also kept?

Comment (numb3r3) [reply annlite/executor.py]: yes, I think so.

File: annlite/executor.py [hunk 3]
```diff
@@ -262,7 +269,7 @@ def search(
                     'Please wait for the pending documents to be indexed.'
                 )
 
-            flat_docs.match(self._index, filter=search_filter, limit=limit)
+            flat_docs.match(self._index, **match_args)
```
Comment (numb3r3) [inline annlite/executor.py:272]: The `filter` is not considered here.

**Review by numb3r3 — CHANGES_REQUESTED** [review]

File: annlite/executor.py (hunk 3 above)
Comment (OrangeSodahub) [reply annlite/executor.py:272]: `filter` is added to `match_args`.

Comment (numb3r3) [reply annlite/executor.py:272]: I see

**Review by numb3r3 — APPROVED** [review]
LGTM

**Pull request merged**